app/feature/feature-movingflow/src/main/kotlin/com/hedvig/android/feature/movingflow/ui/start/StartViewModel.kt

```diff
@@ -0,0 +1,120 @@
+package com.hedvig.android.feature.movingflow.ui.start
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import arrow.core.raise.either
+import arrow.core.raise.ensureNotNull
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.apollo.ErrorMessage
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.feature.movingflow.data.HousingType
+import com.hedvig.android.feature.movingflow.storage.MovingFlowRepository
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.DismissStartError
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.NavigatedToNextStep
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.SelectHousingType
+import com.hedvig.android.feature.movingflow.ui.start.StartEvent.SubmitHousingType
+import com.hedvig.android.feature.movingflow.ui.start.StartUiState.StartError
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import octopus.feature.movingflow.MoveIntentV2CreateMutation
+
+internal class StartViewModel(
+  apolloClient: ApolloClient,
+  movingFlowRepository: MovingFlowRepository,
+) : MoleculeViewModel<StartEvent, StartUiState>(
+    StartUiState.Content(HousingType.entries, HousingType.entries.first(), null, null),
+    StartPresenter(apolloClient, movingFlowRepository),
+  )
+
+private class StartPresenter(
+  private val apolloClient: ApolloClient,
+  private val movingFlowRepository: MovingFlowRepository,
+) : MoleculePresenter<StartEvent, StartUiState> {
+  @Suppress("NAME_SHADOWING")
+  @Composable
+  override fun MoleculePresenterScope<StartEvent>.present(lastState: StartUiState): StartUiState {
+    var selectedHousingType by remember {
+      mutableStateOf((lastState as? StartUiState.Content)?.selectedHousingType ?: HousingType.entries.first())
+    }
+    var submittingHousingType: HousingType? by remember { mutableStateOf(null) }
+    var error: StartError? by remember { mutableStateOf(null) }
+    var moveIntentId: String? by remember { mutableStateOf(null) }
+
+    CollectEvents { event ->
+      when (event) {
+        is SelectHousingType -> selectedHousingType = event.housingType
+        SubmitHousingType -> submittingHousingType = selectedHousingType
+        NavigatedToNextStep -> moveIntentId = null
+        DismissStartError -> error = null
+      }
+    }
+
+    val submittingHousingTypeValue = submittingHousingType
+    if (submittingHousingTypeValue != null) {
+      LaunchedEffect(submittingHousingTypeValue) {
```

Comment:
we won't show loading here?

Reply:
Yeah if `val submittingHousingType: HousingType?,` is not null it should be interpreted as that we are loading the next step here.
I will add a `isLoading: Boolean` value in the data class which is derived from the value of `submittingHousingType` to make this more clear since now it's not really clear at all.